File: Silk4J/src/sample/insurance/LowLevelKeywords.java
/**
 * Copyright 2013 Micro Focus. All rights reserved. 
 * Portions Copyright 1992-2009 Borland Software Corporation (a Micro Focus company).
 */
package sample.insurance;

import org.junit.Assert;

import com.borland.silk.keyworddriven.annotations.Argument;
import com.borland.silk.keyworddriven.annotations.Keyword;
import com.borland.silk.keyworddriven.annotations.KeywordGroup;
import com.borland.silktest.jtf.Desktop;
import com.borland.silktest.jtf.TestObject;
import com.borland.silktest.jtf.xbrowser.DomElement;
import com.borland.silktest.jtf.xbrowser.DomListBox;
import com.borland.silktest.jtf.xbrowser.DomTextField;

@KeywordGroup("API")
public class LowLevelKeywords {
	private Desktop desktop = new Desktop();

	@Keyword("click")
	public void click(@Argument("Locator or ObjectMapID") String identifier) {
		desktop.<DomElement> find(identifier).click();
	}

	@Keyword("setText")
	public void setText(@Argument("Locator or ObjectMapID") String identifier, @Argument("text") String text) {
		TestObject obj = desktop.find(identifier);
		if (obj instanceof DomTextField)
			desktop.<DomTextField> find(identifier).setText(text);
		else
		desktop.<DomElement> find(identifier).typeKeys(text);
	}

	@Keyword("validate")
	public void validate(@Argument("Locator or ObjectMapID") String identifier, @Argument("property") String property, @Argument("expected results") String expectedValue) {
		Assert.assertEquals(expectedValue, desktop.<DomElement> find(identifier).getProperty(property));
	}

	@Keyword("select")
	public void select(@Argument("Locator or ObjectMapID") String identifier, @Argument("item") String item) {
		TestObject obj = desktop.find(identifier);
		if (obj instanceof DomListBox) {
			((DomListBox) obj).select(item);
		} else {
			obj.getParent().<DomElement> find("/IMG").click();
			desktop.<DomElement> find("//DIV[@textContents='" + item + "']").click();
		}
	}
}

File: Silk4NET/LowLevelKeywords.cs
/**
 * Copyright 2013 Micro Focus. All rights reserved. 
 * Portions Copyright 1992-2009 Borland Software Corporation (a Micro Focus company).
 */

using Silk.KeywordDriven;
using SilkTest.Ntf;
using SilkTest.Ntf.XBrowser;

namespace InsuranceWeb
{
    public class LowLevelKeywords
    {
        [Keyword("Start default browser")]
        public void startBrowser()
        {
            Agent.ExecuteBaseState(new BrowserBaseState());
        }

        [Keyword("Start specific browser")]
        public void startBrowser(BrowserType browserType, string url)
        {
            Agent.ExecuteBaseState(new BrowserBaseState(browserType, url));
        }

        [Keyword("Set text")]
        public void SetText(string locator, string text)
        {
            var obj = Agent.Desktop.Find(locator);
            if (obj is DomTextField)
            {
                ((DomTextField)obj).SetText(text);
            }
        }

        [Keyword("Click")]
        public void Click(string locator)
        {
            var obj = Agent.Desktop.Find(locator);
            if (obj is IClickable)
            {
                ((IClickable)obj).Click();
            }
        }
        [Keyword()]
        public void Select(string locator, string value)
        {
            var obj = Agent.Desktop.Find(locator);
            if (obj is DomListBox)
            {
                ((DomListBox)obj).Select(value);
            }
        }

        [Keyword("Close browser")]
        public void CloseBrowser()
        {
            var obj = Agent.Desktop.Find("//BrowserApplication");

            if (obj is BrowserApplication)
            {
                ((BrowserApplication)obj).CloseSynchron();
            }
        }
    }
}
